### PR TITLE
Sanitize 0x7F in Content-Disposition filename

### LINKED ISSUE #47027: Sanitize 0x7F for content disposition header value
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

The issue is when we returned a FileStreamResult got an exception:
```
System.InvalidOperationException: Invalid control character in header: 0x7f
   at Microsoft.AspNetCore.HttpSys.Internal.HeaderCollection.ValidateHeaderCharacters(String headerCharacters)
   at Microsoft.AspNetCore.HttpSys.Internal.HeaderCollection.ValidateHeaderCharacters(StringValues headerValues)
   at Microsoft.AspNetCore.HttpSys.Internal.HeaderCollection.set_Item(String key, StringValues value)
   at Microsoft.AspNetCore.Http.IHeaderDictionary.set_ContentDisposition(StringValues value)
   at Microsoft.AspNetCore.Internal.FileResultHelper.SetContentDispositionHeader(HttpContext httpContext, FileResultInfo& result)
   at Microsoft.AspNetCore.Internal.FileResultHelper.SetHeadersAndLog(HttpContext httpContext, FileResultInfo& result, Nullable`1 fileLength, Boolean enableRangeProcessing, Nullable`1 lastModified, EntityTagHeaderValue etag, ILogger logger)
   at Microsoft.AspNetCore.Mvc.Infrastructure.FileResultExecutorBase.SetHeadersAndLog(ActionContext context, FileResult result, Nullable`1 fileLength, Boolean enableRangeProcessing, Nullable`1 lastModified, EntityTagHeaderValue etag)
   at Microsoft.AspNetCore.Mvc.Infrastructure.FileStreamResultExecutor.ExecuteAsync(ActionContext context, FileStreamResult result)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeResultAsync>g__Logged|22_0(ResourceInvoker invoker, IActionResult result)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|30_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
--- End of stack trace from previous location ---
```

This is because we specify the filename, and there is a control char `0x7F` in the filename.

I checked the code for setup content disposition header value, there is a sanitize operation. Santize operation replaces chars less than 0x20 and large than 0x7f with "_". `0x20` and `0x7F` will not be replaced.
https://github.com/dotnet/aspnetcore/blob/5d10dceeaf5d1d4ac945356f580cce5cf0486c40/src/Http/Headers/src/ContentDispositionHeaderValue.cs#L502

But when validating header value, `0x20` is in allowed chars, `0x7F` is not. https://github.com/dotnet/aspnetcore/blob/5d10dceeaf5d1d4ac945356f580cce5cf0486c40/src/Shared/HttpSys/RequestProcessing/HeaderCollection.cs#L271

### Describe the solution you'd like

I'd like to suggest santize `0x7F` for setup content dispostion value.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amcasey**
We were using the ASCII boundaries, but it looks like we should have been using the ISO-8859 boundaries.

Fixes #47027

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]

**Comment by adityamandaleeka** [comment]
Is there similar code in the runtime repo that should be updated?

**Comment by amcasey** [comment]
@adityamandaleeka Where would I look?

**Comment by adityamandaleeka** [comment]
Not sure, but I just looked at https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/Headers/ContentDispositionHeaderValue.cs and nothing jumped out as needing updating.

**Comment by adityamandaleeka** [comment]
cc @MihaZupan

**Comment by amcasey** [comment]
> Not sure, but I just looked at https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/Headers/ContentDispositionHeaderValue.cs and nothing jumped out as needing updating.

AFAICT, that type doesn't replace any characters with `_` - it just quotes the whole thing if it sees control characters.

File: src/Http/Headers/test/ContentDispositionHeaderValueTest.cs [hunk 1]
```diff
@@ -238,11 +238,16 @@ public void FileNameStar_UnknownOrBadEncoding_PropertyFails()
     [InlineData("FileName.bat", "FileName.bat")]
     [InlineData("FileÃName.bat", "File_Name.bat")]
     [InlineData("File\nName.bat", "File_Name.bat")]
+    [InlineData("File\x19Name.bat", "File_Name.bat")]
```
Comment (amcasey) [inline src/Http/Headers/test/ContentDispositionHeaderValueTest.cs:241]: This should be `\x1F`. 🤦

**Pull request merged**